server/src/com/thoughtworks/go/config/update/UpdateEnvironmentCommand.java

```diff
@@ -32,17 +29,17 @@
 public class UpdateEnvironmentCommand extends EnvironmentCommand implements EntityConfigUpdateCommand<EnvironmentConfig> {
 
     private GoConfigService goConfigService;
-    private final EnvironmentConfig oldEnvironmentConfig;
+    private String oldEnvironmentName;
     private final EnvironmentConfig newEnvironmentConfig;
     private final Username username;
     private String md5;
     private EntityHashingService hashingService;
     private final HttpLocalizedOperationResult result;
 
-    public UpdateEnvironmentCommand(GoConfigService goConfigService, EnvironmentConfig oldEnvironmentConfig, EnvironmentConfig newEnvironmentConfig, Username username, Localizable.CurryableLocalizable actionFailed, String md5, EntityHashingService hashingService, HttpLocalizedOperationResult result) {
+    public UpdateEnvironmentCommand(GoConfigService goConfigService, String oldEnvironmentName, EnvironmentConfig newEnvironmentConfig, Username username, Localizable.CurryableLocalizable actionFailed, String md5, EntityHashingService hashingService, HttpLocalizedOperationResult result) {
         super(actionFailed, newEnvironmentConfig, result);
         this.goConfigService = goConfigService;
-        this.oldEnvironmentConfig = oldEnvironmentConfig;
+        this.oldEnvironmentName = oldEnvironmentName;
```

Comment:
Yes, We support renaming of environments, 
For Finding the old environment, we need `oldEnvironmentName`